### DISCUSSION TITLE
Error: Only one identification found: 'RNA_snn_res.0.6

### DISCUSSION THREAD

**Discussion by Ankita-1211** [reactome/ReactomeGSA#35] [Q&A] [answered]
Hi

I am trying to run reactomeGSA for scRNA seurat result but I am getting this error.

**Error: Only one identification found: 'RNA_snn_res.0.6'. Please ensure that cell / cluster ids are stored as the primary identification (Ident) of your Seurat object. Clustering has to be performed prior to this pathway analysis**.

I know its something related to seurat pipeline, but I have less understanding of this error.

I got another error as well when I just not used this param for my seurat run Idents(seuset) <- "RNA_snn_res.0.6"

**In ReactomeGSA::perform_reactome_analysis(request, verbose = verbose) :
  Reactome Analysis failed: Failed to map any submitted identifiers**

**Comment by jgriss** [comment]
Hi @Ankita-1211,

Thanks a lot for your interest in ReactomeGSA!

You need to store your cluster assignments / cell type assignments as the primary `Ident` in your Seurat object. Otherwise, you don't get any helpful results.

We had a more detailed discussion about this here: #30 

About the identifiers: Which identifier system did you use in your alignment? You can quickly get an example of your identifiers by running:

```R
head(rownames(your_seurat_object))
```

Kind regards,
Johannes

**Reply by Ankita-1211** [reply]
Thanks, Johannes for your reply.

The tool is really helpful for scRNA pathway analysis, it does pretty amazing work.
I am facing a few roadblocks cuz of my limited understanding of seurat pipeline as such.

```head(rownames(your_seurat_object))``` this shows ensembl ids "ENSG00000000003.14", "ENSG00000000419.12"                "ENSG00000000457.13"

is it not supported ? that's the reason for Reactome Analysis failed: Failed to map any submitted identifiers.

here is the snapshot of my seurat count data input.

<html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">
<head>

<meta name=Generator content="Microsoft Excel">

<style>

</style>
</head>
<body>



gene | gene_name | AAACCCAAGTGGTCAG | AAAGGTATCAACTACG | AAAGTCCAGCGTGTCC | AACACACTCAAGAGTA
-- | -- | -- | -- | -- | --
ENSG00000000003.14 | TSPAN6 | 0 | 0 | 0 | 0
ENSG00000000419.12 | DPM1 | 0 | 0 | 1 | 0
ENSG00000000457.13 | SCYL3 | 0 | 0 | 0 | 0
ENSG00000000457.13,ENSG00000000460.16 | NA | 0 | 0 | 0 | 0
ENSG00000000460.16 | C1orf112 | 0 | 0 | 0 | 0
ENSG00000000460.16,ENSG00000000457.13 | NA | 0 | 0 | 0 | 0
ENSG00000000938.12 | FGR | 1 | 0 | 0 | 1
ENSG00000000971.15 | CFH | 0 | 0 | 0 | 0
ENSG00000001036.13 | FUCA2 | 0 | 0 | 0 | 0
ENSG00000001084.12 | GCLC | 0 | 0 | 0 | 0
ENSG00000001084.12,ENSG00000249379.1 | NA | 0 | 0 | 0 | 0
ENSG00000001167.14 | NFYA | 0 | 0 | 0 | 0
ENSG00000001167.14,ENSG00000124596.16 | NA | 0 | 0 | 0 | 0



</body>

</html>

**Comment by jgriss** [comment, marked as answer]
Hi @Ankita-1211,

The issue is caused by the way the alignment was done.

1) Reactome does not support version numbers in ENSEMBL ids (the number after the ".")

This can easily be fixed using the following command:

```R
# Remove any trailing .[Number] from the identifiers
rownames(your_seurat_object) <- gsub("\\.[0-9]*$", "", rownames(your_seurat_object))
```

2) Apparently, your pipeline created groups of ambiguous genes (f.e. "ENSG00000001084.12,ENSG00000249379.1"). Naturally, these cannot be interpreted by our tools since we don't know which gene to refer to. It won't harm your analysis as they will simply be ignored.

Kind regards,
Johannes

**Reply by Ankita-1211** [reply]
Thank you so much. Your suggestions were really very helpful.
